src/packages.jl

```diff
@@ -64,3 +64,59 @@ function LoadPackageAndExposeGlobals(package::String, mod::Module; all_globals::
 end
 
 export LoadPackageAndExposeGlobals
+
+
+module Packages
+
+import ...GAP: Globals, julia_to_gap
+
+"""
+    load(desc)
+
+TODO: write text, link to GAP's LoadPackage
+
+TODO: support version qualifier?
+"""
+function load(desc::String)
+    return Globals.LoadPackage(julia_to_gap(desc), false)
+    # TODO: can we provide more information in case of a failure?
+    # GAP unfortunately only gives us info messages...
+end
```

Comment:
Concerning the question about more information, what is wrong with info messages?
The typical advice in GAP is to call `SetInfoLevel(InfoPackageLoading,4);` and then to call `LoadPackage` again.

Concerning optional arguments of `load`, also the flag for showing/suppressing package banners could be supported.

Reply:
I think the current behavior of GAP is very user unfriendly, already on the GAP side. There is no reason (other than technical ones, i.e., "nobody implemented this") why `LoadPackage("foo")` when entered interactively couldn't immediately tell the user things like: "can't load package `foo` because its dependency `bar` is missing", or "can't load `foo` because it require `bar`, and that requires `qux`, which is missing". And instead of "missing" of course we could also have other error reasons.

Anyway, that's not even what I had in mind here: point is that LoadPackage could fail because the package is not present at all; or because it failed to load (e.g. wasn't compiled, or a required external dependency is missing). Depending on which it is, the user might want to do different things to resolve it.

Reply:
Yes, even with `SetInfoLevel(InfoPackageLoading,4)` it is always a non-trivial task to see what is missing, the first challenge being the deciphering of the long, long output. And even with an experienced eye, one has to do this exercise several times with almost to further help from GAP until one gets all dependencies satisfied. Julia is really a dream in this regard: `] add XYZ`, that's it.

Reply:
We must distinguish between installing and loading GAP packages.
The task to get the dependencies satisfied belongs to the installation; this belongs on the Julia side to Julia's `add`, and on the GAP side it was assumed to happen outside a GAP session until the idea of a package manager came up.
GAP's `LoadPackage` corresponds to Julia's `using`; in particular, `LoadPackage` should never attempt to change anything in the user's GAP installation.

(Of course I do not object to getting better information from `LoadPackage` or `TestPackageAvailability`.)